samza-core/src/main/java/org/apache/samza/table/remote/RemoteReadWriteTable.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.table.remote;
+
+import java.util.List;
+import java.util.function.Function;
+
+import org.apache.samza.SamzaException;
+import org.apache.samza.container.SamzaContainerContext;
+import org.apache.samza.metrics.Counter;
+import org.apache.samza.metrics.Timer;
+import org.apache.samza.operators.KV;
+import org.apache.samza.storage.kv.Entry;
+import org.apache.samza.table.ReadWriteTable;
+import org.apache.samza.task.TaskContext;
+import org.apache.samza.util.RateLimiter;
+
+import com.google.common.base.Preconditions;
+
+import static org.apache.samza.table.remote.RemoteTableDescriptor.RL_WRITE_TAG;
+
+
+/**
+ * Remote store backed read writable table
+ *
+ * @param <K> the type of the key in this table
+ * @param <V> the type of the value in this table
+ */
+public class RemoteReadWriteTable<K, V> extends RemoteReadableTable<K, V> implements ReadWriteTable<K, V> {
+  protected final TableWriteFunction<K, V> writeFn;
+  protected final Function<KV<K, V>, Integer> writeCreditFn;
```

Comment:
I actually thought the sole existence of CreditFunction<K,V> instead of the plain Function<K,V> is to support lambda. Otherwise we can just use Function<K,V>. But I don't see any issues making CreditFunction<K,V> a top-level interface and use it in both the descriptor and ReadableTable/ReadWriteTable.